tests/functional/api/retry.js

```diff
@@ -0,0 +1,678 @@
+const assert = require('assert');
+const async = require('async');
+const http = require('http');
+const querystring = require('querystring');
+const { RedisClient } = require('arsenal').metrics;
+const { StatsModel } = require('arsenal').metrics;
+
+const config = require('../../config.json');
+const getUrl = require('../utils/getUrl');
+const { addMembers, addManyMembers } = require('../utils/sortedSetHelpers');
+const { makeRetryPOSTRequest, getRequest, getResponseBody } =
+    require('../utils/httpHelpers');
+const fakeLogger = require('../utils/fakeLogger');
+
+const redisConfig = { host: '127.0.0.1', port: 6379 };
+const redisClient = new RedisClient(redisConfig, fakeLogger);
+const useRoleAuth = process.env.CI_AUTH_TYPE === 'role';
```

Comment:
Now maybe it could make more sense to check the "authType" specified in the config rather than rely on an env var.

Reply:
Good call, updated.